consensus/src/detail/test.rs

```diff
@@ -1423,3 +1423,51 @@ fn test_events_orphan_block() {
         assert!(consensus.process_block(block, BlockSource::Local).is_err());
     });
 }
+
+#[test]
+fn test_get_locator() {
+    common::concurrency::model(|| {
+        let config = create_mainnet();
+        let storage = Store::new_empty().unwrap();
+        let mut consensus = Consensus::new_no_genesis(config, storage).unwrap();
+
+        let block_source = BlockSource::Local;
+        let mut block_index = consensus
+            .process_block(consensus.chain_config.genesis_block().clone(), block_source)
+            .ok()
+            .flatten()
+            .expect("Unable to process genesis block");
+
+        let mut prev_block = consensus.chain_config.genesis_block().clone();
+        let limit = rand::thread_rng().gen::<u16>();
+
+        for _ in 0..limit {
+            let block_source = BlockSource::Peer(1);
+            let new_block = produce_test_block(&consensus.chain_config, &prev_block, false);
+            let new_block_index =
+                consensus.process_block(new_block.clone(), block_source).ok().flatten().unwrap();
+
+            block_index = new_block_index;
+            prev_block = new_block;
+        }
+        let locator = consensus.get_locator().unwrap();
+
+        // only genesis
+        if limit == 0 {
+            assert_eq!(locator.len(), 1);
+        } else {
+            assert_eq!(locator.len(), (limit as f64).log2().floor() as usize + 2);
```

Comment:
Out of curiosity, why do we need log2 here?

Reply:
The locator is like a function `fn(n) = get_block_at(2^n)` so it collects the blocks 1,2,4,8,16,32... Taking `floor(log2(our_height))` should give the length of the locator (+ 2) if everything works as it should/I've understood locators correctly.